File: problems/58-length-of-last-word.ts
function lengthOfLastWord(s: string): number {
  let start = 0
  let end: number | undefined

  for (let i = s.length - 1; i >= 0; i--) {
    if (end === undefined && s[i] !== " ") {
      end = i
    } else if (end !== undefined && s[i] === " ") {
      start = i + 1
      break
    }
  }

  return end === undefined ? 0 : end - start + 1
}

;(() => {
  const testCases = [
    "Hello World",
    "   fly me   to   the moon  ",
    "luffy is still joyboy",
  ]

  testCases.forEach((tc) =>
    console.log(
      `\nInput: ${JSON.stringify(tc)}\nOutput: ${lengthOfLastWord(tc)}`
    )
  )
})()

File: problems/58_length_of_last_word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        n = len(s)
        cur = n - 1
        while s[cur] == " ":
            cur -= 1
        length = 0
        while cur >= 0 and s[cur] != " ":
            length += 1
            cur -= 1
        return length
